File: modules/@angular/platform-browser/test/browser/tools/spies.ts
import {ReflectiveInjector, provide} from '@angular/core';
import {ApplicationRef, ApplicationRef_} from '@angular/core/src/application_ref';
import {SpyObject} from '@angular/core/testing/testing_internal';

import {global} from '../../../src/facade/lang';

export class SpyApplicationRef extends SpyObject {
  constructor() { super(ApplicationRef_); }
}

export class SpyComponentRef extends SpyObject {
  injector: any /** TODO #9100 */;
  constructor() {
    super();
    this.injector = ReflectiveInjector.resolveAndCreate(
        [{provide: ApplicationRef, useClass: SpyApplicationRef}]);
  }
}

export function callNgProfilerTimeChangeDetection(config?: any /** TODO #9100 */): void {
  (<any>global).ng.profiler.timeChangeDetection(config);
}

File: modules/@angular/platform-browser/test/browser/tools/spies.dart
import 'package:angular2/testing_internal.dart' show SpyObject;
import 'package:angular2/core.dart' show Injector, ReflectiveInjector, bind;
import 'package:angular2/src/core/application_ref.dart' show ApplicationRef;
import 'package:angular2/src/core/linker/component_factory.dart'
    show ComponentRef;
import 'dart:js';

@proxy
class SpyApplicationRef extends SpyObject implements ApplicationRef {
  tick() {}
}

@proxy
class SpyComponentRef extends SpyObject implements ComponentRef<dynamic> {
  Injector injector;

  SpyComponentRef() {
    this.injector = ReflectiveInjector
        .resolveAndCreate([{provide: ApplicationRef, useClass: SpyApplicationRef}]);
  }
}

void callNgProfilerTimeChangeDetection([config]) {
  context['ng']['profiler']
      .callMethod('timeChangeDetection', config != null ? [config] : []);
}
